prometheus_hardware_exporter/core.py

```diff
@@ -96,14 +96,18 @@ def failed_metrics(self) -> Iterable[Metric]:
         Yields:
             metrics: the internal metrics
         """
-        name = self.__class__.__name__
+        # if `__class_.__name__` is "IpmiDcmiCollector"
+        # then `name` becomes "ipmidcmi"
+        suffix = "collector"
+        class_name = self.__class__.__name__.lower()
+        name = class_name[: -len(suffix)] if class_name.endswith(suffix) else class_name
```

Comment:
This does remove "collector" anywhere in the string as opposed to just the suffix which was what I originally intended. But I think that's alright for our use case and it does indeed make the code cleaner. Thanks for your suggestion, I'll make the change along with the helpful comment.

Reply:
If you want you can use `.removesuffix` functions, if there is possibility that `collector` will in in middle or begging of name.

```python
>>> a = "ipmidcmicollector"
>>> a.removesuffix("collector")
'ipmidcmi'
```

Reply:
My first idea was to use the [removesuffix](https://docs.python.org/3/library/stdtypes.html#str.removesuffix) function but unfortunately it but it was only introduced in Python 3.9. And as part of our CI tests, we also test on Python 3.8. So it wouldn't work for us :(

Reply:
In that case I vote for using replace, instead of indexing `class_name[: -len(suffix)]` or `rsplit` or you want you can add simple function to remove only from right side. 

```python
>>> name, _ = "collectornicecollector".rsplit("collector", 1)
>>> name
'collectornice'
```